python/beeai_framework/agents/base.py

```diff
@@ -25,12 +25,12 @@
 from beeai_framework.memory import BaseMemory
 from beeai_framework.utils.models import ModelLike
 
-T = TypeVar("T", bound=BaseModel)
 RI = TypeVar("RI", bound=BaseModel)
 RO = TypeVar("RO", bound=BaseModel)
+OUT = TypeVar("OUT", bound=BaseModel)
```

Comment:
The naming seems quite akward to me.

What about?

```python
TInput = TypeVar("TInput", bound=BaseModel)
TOptions = TypeVar("TOptions", bound=BaseModel)
TOutput = TypeVar("TOutput", bound=BaseModel)
```

Reply:
I used the named we used in Tool, but your match the TS implementation so I can make the update to match TS. and I'll update Tool as well later when I get to it's my issues